rcl/src/rcl/wait.c

```diff
@@ -430,6 +436,17 @@ rcl_wait(rcl_wait_set_t * wait_set, int64_t timeout)
     &dummy_clients,
     wait_set->impl->rmw_waitset,
     timeout_argument);
+  // Check for timeout.
+  if (ret == RMW_RET_TIMEOUT) {
+    // Assume none were set (because timeout was reached first), and clear all.
+    rcl_ret_t rcl_ret;
+    (void)rcl_ret;  // NO LINT
```

Comment:
Because in release mode the `assert` statements are gone and without this you get a "set but never used".